### DISCUSSION TITLE
cannot pull docker container sha256 error on f1f00ef4e9ed layer

### DISCUSSION THREAD

**Discussion by TeaCult** [idaholab/moose#22788] [Q&A Getting Started] [answered]
## Bug Description
I cannot run or pull container strage error occurs : 

docker run -ti idaholab/moose:latest /bin/bash -c 'cd test; ./run_tests'
Unable to find image 'idaholab/moose:latest' locally
latest: Pulling from idaholab/moose
eaead16dc43b: Pull complete 
4f4fb700ef54: Pull complete 
4a19a462196d: Pull complete 
68809389b6a7: Pull complete 
9bcd8b69b17e: Pull complete 
f29b50332647: Pull complete 
b67fa59dd2d9: Pull complete 
3bf93d7f56e4: Pull complete 
f1f00ef4e9ed: Extracting [==================================================>]  1.588kB/1.588kB
783d9607c2be: Download complete 
30af6333143c: Download complete 
c7d674dc8468: Download complete 
22494a5ceeed: Download complete 
a9e2d0f1c181: Download complete 
7a1756df8f03: Download complete 
072d860238cd: Download complete 
19290591cfbb: Download complete 
4c2d687235a3: Download complete 
docker: failed to register layer: rename /var/lib/docker/image/btrfs/layerdb/tmp/write-set-2670145948 /var/lib/docker/image/btrfs/layerdb/sha256/3cbfecd5b748c8e7a47ca492d03f26007fe50dc5a5cec360928685d6d074c8b7: file exists.
See 'docker run --help'.

## Steps to Reproduce
docker run -ti idaholab/moose:latest /bin/bash -c 'cd test; ./run_tests'

## Impact
cannot use moose in docker container 

## My configuration: 

I am on a rtx 3060 passed through vm which has lxd containers. In one of lxd image called fem I have docker containers:
REPOSITORY             TAG       IMAGE ID       CREATED         SIZE
dolfinx/lab            stable    bab989bbd385   2 months ago    3.92GB
dealii/dealii          latest    3921e1752cbb   4 months ago    6.28GB
uwsbel/projectchrono   latest    38eb2b51feee   10 months ago   7.98GB
and petsce precice calculix and elmerfem built from sources. Everything works except moose because I cant pull it. 

All my base system VM and lxd containers have kernel 6.0.7-arch1 
Base OS [vfio-kvm [ lxc [ docker [ !mose_image] ] ] ]

**Comment by TeaCult** [comment]
rebooted 3 time and now all works. I really have no idea whats happening , lots of free disk space , storage pool space ram and cpu resources are available. Check sha256 :

```
ls /var/lib/docker/image/btrfs/layerdb/sha256/3cbfecd5b748c8e7a47ca492d03f26007fe50dc5a5cec360928685d6d074c8b7
cache-id  diff	parent	size  tar-split.json.gz
```


Cleaning cache also didnt work. 
Strangely after 3 reboots , everything is normal:


```
docker pull idaholab/moose:latest 
latest: Pulling from idaholab/moose
eaead16dc43b: Pull complete 
4f4fb700ef54: Pull complete 
8198d4bd982d: Pull complete 
27f4ad5165da: Pull complete 
fbd732637927: Pull complete 
75844ee65ec2: Pull complete 
c6e9c5625073: Pull complete 
9724f9c73dd7: Pull complete 
ea7ed87a478a: Pull complete 
1bb1f22ca993: Pull complete 
1ca59dfea254: Pull complete 
858a9e363486: Pull complete 
5f9aedc8ada4: Pull complete 
51fb2c57dcdc: Pull complete 
247f4d4cc6bb: Pull complete 
662520e63b9e: Pull complete 
dc359c013330: Pull complete 
48b4b0faf25c: Pull complete 
Digest: sha256:65b83ed0e07c1f5de2be415b03eb25d887514261a30865a64e281534389e4a3c
Status: Downloaded newer image for idaholab/moose:latest
docker.io/idaholab/moose:latest
```

**Comment by loganharbour** [comment, marked as answer]
I've actually had this issue before - luckily it's all on docker and not on us :-) 

If you search for that error string, you can see lots of complaints about docker cache getting screwed up in time. I usually get away with doing a prune and then removing the offending cache file, but it seems like you've got it working for now.
